src/App/App.jsx

```diff
@@ -0,0 +1,40 @@
+import React, { useState } from 'react'
+import { GraphiQL } from 'graphiql'
+import { createGraphiQLFetcher } from '@graphiql/toolkit'
+
+import { federationInfoPlugin } from '../federation-info-plugin'
+
+import 'graphiql/graphiql.css'
+import './App.scss'
+
+function App() {
+  const [hideSecondPanel, setHideSecondPanel] = useState(false)
+
+  const federationSchemaUrl = 'http://localhost:3001/federation-schema'
+
+  const fetcher = createGraphiQLFetcher({
+    url: 'http://localhost:3001/graphql'
+  })
+
+  return (
+    <div
+      className={
+        hideSecondPanel ? 'full-height hide-second-panel' : 'full-height'
+      }
+    >
+      <GraphiQL
+        fetcher={fetcher}
+        plugins={[federationInfoPlugin({ federationSchemaUrl })]}
+        onTogglePluginVisibility={plugin => {
```

Comment:
I don't think this approach is ok, unless we change how mercurious implements graphiql plugins.

See here: <https://github.com/mercurius-js/mercurius/blob/a230f81872418603874cdb7f4ac110be0231f353/static/main.js#L121>

The problem is that mercurious only adds plugins to `plugins=` attribute. So ideally would be if we can trigger that panel hiding from the plugin itself (through context maybe?)

Reply:
Hmm yes you are right. As discussed we will use this PR to incorporate MUI and a future PR for the full screen mode.